wazo_webhookd/auth.py

```diff
@@ -21,59 +28,65 @@ def __init__(self):
         super().__init__(503, msg, 'master-tenant-not-initialized')
 
 
+def get_auth_token_from_request() -> str | None:
+    return request.headers.get('X-Auth-Token') or request.args.get('token')
+
+
 class Token:
-    def __init__(self, auth_client, token_id):
+    def __init__(self, auth_client: AuthClient, token_id: str | None) -> None:
         try:
-            self._token_infos = auth_client.token.get(token_id)
+            self._token_info: TokenDict = auth_client.token.get(token_id)
         except HTTPError as e:
             logger.warning('HTTP error from wazo-auth while getting token: %s', e)
             raise AuthServerUnreachable(auth_client.host, auth_client.port, e)
 
-    def user_uuid(self):
-        user_uuid = self._token_infos['metadata']['uuid']
-        if not user_uuid:
-            raise TokenWithUserUUIDRequiredError()
-        return user_uuid
+    @property
+    def user_uuid(self) -> str:
+        if user_uuid := self._token_info['metadata']['uuid']:
+            return user_uuid
+        raise TokenWithUserUUIDRequiredError()
```

Comment:
Oh really? I've seen it very often. Normally they only raise errors if there is something really unexpected, but it happens. It's mostly for comprehension. The name implies and attribute and not a method. `myclass.user_uuid()` just seems strange, normally it would be `myclass.get_user_uuid()` or `myclass.user_uuid`. If we know it doesn't change on an instance it's also a good opportunity to use `cached_property` to speed things up.

Reply:
OK, thanks for the explanation. No objections to keep it like this.